docs/EXTENSIONS_GUIDE.md

```diff
@@ -0,0 +1,53 @@
+# Lisk Hub Extension Guideline
+``` ⚠️ Extension System is only avaliable in development mode ⚠️```
+
+## How to create a new extension?
+In your extension, you will have access to `React` and `LiskHubExtensions` that allows adding Your extension to specified areas in our code.
+
+![Alt text](./assets/create_extension_1.png?raw=true "Github File with extension")
+
+For building Your Component, You can use es6 and our wrapper will transpaile it for You.
+
+![Alt text](./assets/create_extension_2.png?raw=true "Github File with extension")
+
+#### LiskHubExtensions
+| Properties          |                    Description                 |
+| --------------- |---------------------------------------------|
+| `addModule({ identifier, component })` | identifier: `String`, component: `Component` <br/> Adds new component to the place we specify               |
+| `components`            | Contains shared styled components that We use in our app.             |
+| `identifiers`            | Identifier you pass to addModule() `dashboardColumn1, dashboardColumn2, dashboardColumn3`. It will add your component below dashboard components in specific column          |
+
+![Alt text](./assets/create_extension_3.png?raw=true "Github File with extension")
+
+#### Props
+Data from our store is accessible via `this.props.data` and actions to modify our store are in `this.props.actions`. Keep in mind that We are not sharing all data and actions. In case You want to access sth that is unavailable let us know by creating a feature proposal/ticket in our repo.
+
+#### How to Share Your extension with others?
```

Comment:
I would put here a note to distinguish between how it works now vs. how it should work in the future.